### PR TITLE
Add more seo things

### PR DESCRIPTION
**Pull request by wafer-li**
Port from https://github.com/wafer-li/hexo-theme-next-reloaded/commit/ec845557e0e34948ef18b72cb9216eb9c934b10e

Add external+nofollow for social and powered when seo on

## PR Checklist
**Please check if your PR fulfills the following requirements:**

- [x] The commit message follows [our guidelines](https://github.com/theme-next/hexo-theme-next/blob/master/.github/CONTRIBUTING.md).
- [x] Tests for the changes have been added (for bug fixes / features).
   - [ ] Muse | Mist have been tested.
   - [x] Pisces | Gemini have been tested.
- [ ] Docs have been added / updated (for bug fixes / features).

## PR Type
**What kind of change does this PR introduce?**  

- [ ] Bugfix.
- [ ] Feature.
- [ ] Code style update (formatting, local variables).
- [ ] Refactoring (no functional changes, no api changes).
- [ ] Build related changes.
- [ ] CI related changes.
- [ ] Documentation content changes.
- [x] Other... Please describe: Enhancement

## What is the current behavior?

The social links and the powered links don't have `rel="external nofollow"`

## What is the new behavior?
Add `rel="external nofollow"` to social links and powered links, when seo enable

### How to use?
In NexT `_config.yml`:
```yml
No change to config
```

## Does this PR introduce a breaking change?
- [ ] Yes.
- [x] No.

### CODE REVIEW

File: layout/_partials/footer.swig [hunk 1]
```diff
@@ -35,8 +35,12 @@
 </div>
 
 {% if theme.footer.powered %}
+  {% set hexo_link_tag = '<a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a>' %}
+  {% if theme.seo %}
+    {% set hexo_link_tag = '<a class="theme-link" target="_blank" href="https://hexo.io" rel="external nofollow">Hexo</a>' %}
+  {% endif %}
   <div class="powered-by">{#
-  #}{{ __('footer.powered', '<a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a>') }}{#
```
Comment (ivan-nginx) [inline layout/_partials/footer.swig]: Can u here do like above, without `hexo_link_tag` variable?

Comment (wafer-li) [reply layout/_partials/footer.swig]: I have no idea about whether the `{% if %}` tags works within the `__()` as a string.

It is not autocompleted by Atom when I type `if` in the string.

Comment (ivan-nginx) [reply layout/_partials/footer.swig]: Ok, i'll see it later, don't merge.

File: layout/_partials/footer.swig [hunk 2]
```diff
@@ -34,9 +34,13 @@
   {% endif %}
 </div>
 
+{% if theme.seo %}
+  {% set nofollow = ' rel="external nofollow"' %}
```
Comment (ivan-nginx) [inline layout/_partials/footer.swig]: Just like this.

Comment (wafer-li) [reply layout/_partials/footer.swig]: OK, I will adopt it in `sidebar.swig`

Comment (ivan-nginx) [reply layout/_partials/footer.swig]: No, there is no needed because there are single conditions.

File: layout/_partials/footer.swig [hunk 3]
```diff
@@ -34,9 +34,13 @@
...
+{% endif %}
+
 {% if theme.footer.powered %}
   <div class="powered-by">{#
-  #}{{ __('footer.powered', '<a class="theme-link" target="_blank" href="https://hexo.io">Hexo</a>') }}{#
+  #}{{ __('footer.powered', '<a class="theme-link" target="_blank"' + nofollow + ' href="https://hexo.io">Hexo</a>') }}{#
```
Comment (ivan-nginx) [inline layout/_partials/footer.swig]: I mean here like this.

Comment (wafer-li) [reply layout/_partials/footer.swig]: OK.
Merge it now.

**Pull request merged**